partiql-tests-data/eval/primitives/basic.ion

```diff
@@ -319,14 +319,25 @@ basic::[
   {
     name:"arithmetic with null/missing",
     statement:"SELECT VALUE 1 + v FROM << null, missing, 1 >> AS v",
-    assert:{
-      evalMode:[EvalModeCoerce, EvalModeError],
-      result:EvaluationSuccess,
-      output:$bag::[
-        null,
-        null,
-        2
-      ]
-    }
+    assert:[
```

Comment:
Is this difference between modes supported by the spec, or based on the kotlin impl ?

Reply:
I'll need to look into this more and check with Almann. The existing behavior in the test was incorrect -- was an error in the script porting tests from the EvaluatorTestSuite.kt. I'm not too certain on the correct behavior for operations involving `NULL`/`MISSING` in the STRICT/ERROR evaluation mode. There was a [spec test](https://github.com/partiql/partiql-tests/blob/d5a2792d07f08f7140c7f32ceb158396aacd9001/partiql-tests-data/eval/spec-tests.ion#L403-L417) I defined that said `5 + MISSING` errors in the STRICT/ERROR evaluation mode, which contradicts the current behavior in the PR (which is more aligned to PartiQL Kotlin's LEGACY mode).

[Section 7](https://partiql.org/assets/PartiQL-Specification.pdf#section.7) of the spec states the behavior with respect to permissive/coerce mode but is less clear on the error mode whether or not unknown operands will cause an error or will propagate unknowns.

Reply:
After discussing with Almann, the behavior in this test and a couple [spec tests](https://github.com/partiql/partiql-tests/blob/d5a2792d07f08f7140c7f32ceb158396aacd9001/partiql-tests-data/eval/spec-tests.ion#L388-L417) was incorrect in the conformance test suite. Arithmetic expressions in both typing modes (coerce/permissive and error/strict) should allow unknown arguments and propagate to the unknown (e.g. `MISSING + 5` results in `MISSING` in both typing modes). The error/strict typing mode only dictates what the result should be in the case of a data type mismatch -- unknown propagation occurs just as in SQL.

So for this particular case, the ported test should return a bag of `null`, `missing`, and `2` in both typing modes. `partiql-lang-kotlin` returns `null`, `null`, and `2` since it doesn't support an error/strict typing mode and the test is run using the `LEGACY` mode.

Reply:
This test behavior is corrected in https://github.com/partiql/partiql-tests/pull/34/commits/10d12ee826c0cb6c91bf69069d9447b3af7d6b01